### ISSUE TITLE
Support for GADTs

### ISSUE THREAD

**Issue by kamoii** [google/ghc-source-gen#63]
Is support for GADTs planned?

**Comment by judah** [comment]
Yes, adding support for GADTs makes a lot of sense.  I haven't looked closely at what an API for them would look like.  If anyone would like to take a try at implementing it, the docs for `HsDataDefn` describe how GADTs are represented differently than regular Haskell-98 datatypes:
http://hackage.haskell.org/package/ghc-8.6.5/docs/HsDecls.html#t:HsDataDefn
